tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/publicAsset/PublicAssetSetService.scala

```diff
@@ -0,0 +1,40 @@
+package com.linagora.tmail.james.jmap.publicAsset
+
+import jakarta.inject.{Inject, Named}
+import org.apache.commons.configuration2.Configuration
+import org.apache.james.jmap.api.model.{IdentityId, UploadId}
+import org.apache.james.jmap.core.AccountId
+import org.apache.james.jmap.mail.BlobId
+import org.apache.james.jmap.method.IdentityResolver
+import org.apache.james.mailbox.MailboxSession
+import reactor.core.scala.publisher.{SFlux, SMono}
+
+import scala.util.Try
+
+class PublicAssetSetService @Inject()(val identityResolver: IdentityResolver,
+                                      @Named("jmap") jmapConfiguration: Configuration) {
+
+  private val publicURIPrefix: String = jmapConfiguration.getString("publicAsset.publicURIPrefix", "")
+  private val uploadIdPrefix = "uploads-"
+
+  def generatePublicURI(publicAssetId: PublicAssetId, accountId: AccountId): PublicURI =
+    PublicURI.from(publicAssetId, accountId, publicURIPrefix)
+
+  def checkIdentityIdsExist(identityIds: Seq[IdentityId], session: MailboxSession): SMono[Seq[IdentityId]] =
+    SFlux.fromIterable(identityIds)
+      .concatMap(identityId => checkIdentityIdExist(identityId, session))
+      .collectSeq()
+
+  private def checkIdentityIdExist(identityId: IdentityId, session: MailboxSession): SMono[IdentityId] =
+    identityResolver.resolveIdentityId(identityId, session)
+      .flatMap(SMono.justOrEmpty)
```

Comment:
.handle + ReactorUtils 'publishIfPresent'

We do not need to mandate an asynchronous pipeline for something that is a synchronous operation...